pootle/apps/virtualfolder/utils.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import os
+from fnmatch import fnmatch
+
+from pootle.core.url_helpers import split_pootle_path
+from pootle_store.models import Store
+
+
+class VirtualFolderPathMatcher(object):
```

Comment:
I don't recall seeing any test for this, despite it is a critical component of vfolder support.